### ISSUE TITLE
UI appears empty, plugin throws exception when builds complete

### ISSUE THREAD

**Issue by brantb** [tcplugins/tcWebHooks#23]
TeamCity version: 9.1.6 
Server OS: Windows Server 2008 R2
Plugin version: 0.9.35.72 (Latest stable release on GitHub), also tested with 0.9.74.76 (latest `master` build on teamcity.jetbrains.com) and 1.1-alpha4.54.105 (latest alpha release on GitHub)
Symptoms: When a build with webhooks defined completes, the following is written to `teamcity-server.log`:
```
[2016-03-16 11:56:40,065]  DEBUG -   jetbrains.buildServer.SERVER - About to process WebHooks for project408 at buildState buildStarted 
[2016-03-16 11:56:40,066]   WARN - ildServer.util.EventDispatcher - Error in event handler: java.lang.IllegalStateException: Unable to create settings for webhooks; corresponding factory was not registered 
[2016-03-16 11:56:40,066]  DEBUG - ildServer.util.EventDispatcher - Error in event handler 
java.lang.IllegalStateException: Unable to create settings for webhooks; corresponding factory was not registered
	at jetbrains.buildServer.serverSide.settings.ProjectSettingsManagerImpl.getFactory(ProjectSettingsManagerImpl.java:47)
	at jetbrains.buildServer.serverSide.settings.ProjectSettingsManagerImpl.getSettings(ProjectSettingsManagerImpl.java:1)
	at webhook.teamcity.WebHookListener.getListOfEnabledWebHooks(WebHookListener.java:135)
	at webhook.teamcity.WebHookListener.processBuildEvent(WebHookListener.java:98)
	at webhook.teamcity.WebHookListener.buildStarted(WebHookListener.java:170)
	at sun.reflect.GeneratedMethodAccessor245.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at jetbrains.buildServer.util.EventDispatcher.dispatch(EventDispatcher.java:120)
	at jetbrains.buildServer.serverSide.ServerSideEventDispatcher.superDispatch(ServerSideEventDispatcher.java:13)
	at jetbrains.buildServer.serverSide.ServerSideEventDispatcher$1.run(ServerSideEventDispatcher.java:1)
	at jetbrains.buildServer.serverSide.impl.auth.SecurityContextImpl.runAs(SecurityContextImpl.java:8)
	at jetbrains.buildServer.serverSide.impl.auth.SecurityContextImpl.runAsSystem(SecurityContextImpl.java:10)
	at jetbrains.buildServer.serverSide.ServerSideEventDispatcher.dispatch(ServerSideEventDispatcher.java:7)
	at jetbrains.buildServer.util.EventDispatcher$2.invoke(EventDispatcher.java:67)
	at com.sun.proxy.$Proxy19.buildStarted(Unknown Source)
	at jetbrains.buildServer.serverSide.impl.BuildStarter$1.run(BuildStarter.java:11)
	at jetbrains.buildServer.serverSide.impl.BuildStarter$2$1.run(BuildStarter.java:1)
	at jetbrains.buildServer.util.NamedThreadFactory.executeWithNewThreadName(NamedThreadFactory.java:89)
	at jetbrains.buildServer.serverSide.impl.BuildStarter$2.run(BuildStarter.java:1)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
The same exception is also thrown when navigating to the 'WebHooks' tab in the TeamCity UI:
```
[2016-03-16 12:52:37,497]   WARN -   jetbrains.buildServer.SERVER - Unhandled exception occurred while calling method fillModel of WebHookProjectTabExtension[myIncludeUrl='/project/webHooks.jsp', myPluginName='webHooks', myPlaceId=/project.html#PROJECT@TAB] extension. 
java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor278.invoke(Unknown Source)
[snip]
Caused by: java.lang.IllegalStateException: Unable to create settings for webhooks; corresponding factory was not registered
	at jetbrains.buildServer.serverSide.settings.ProjectSettingsManagerImpl.getFactory(ProjectSettingsManagerImpl.java:47)
	at jetbrains.buildServer.serverSide.settings.ProjectSettingsManagerImpl.getSettings(ProjectSettingsManagerImpl.java:1)
	at webhook.teamcity.extension.WebHookProjectTabExtension.fillModel(WebHookProjectTabExtension.java:63)
	at jetbrains.buildServer.web.openapi.project.ProjectTab.fillModel(ProjectTab.java:60)
	... 50 more
```
Unfortunately, I have no idea when this started happening -- I set up the webhooks plugin months ago when we were still running TC8 and I didn't notice that it had stopped working until today.

**Comment by netwolfuk** [comment]
Thanks for your bug report, and especially for taking the time to enable debug and test so many versions. I'll try TeamCity 9.1.6 (I must admit I'm still running 9.1.3) and try to recreate your issue.

In the mean time, you could try one of these dev builds today before I get a chance to look at it (hopefully before the weekend).
[0.9.67.80](https://teamcity.jetbrains.com/repository/download/WebHooksAndOtherPlugins_TcWebHooksBranch09xX/719511:id/tcWebHooksPlugin-0.9.67.80.zip)
[1.1.68.119](https://teamcity.jetbrains.com/repository/download/WebHooksAndOtherPlugins_TcWebHooksBranch09xX/721553:id/tcWebHooksPlugin-1.1.68.119.zip)

**Comment by netwolfuk** [comment]
I'm not able to reproduce this problem in TeamCity 9.1.6.

Could you send (via email - this username at the gmails) or post (as a comment) the webhooks element from your <project>/pluginData/plugin-settings.xml file.

It probably contains API keys or something so maybe remove anything from the URLs that you don't want to share with me/us.

**Comment by brantb** [comment]
Turns out this was being caused by a conflict with a different TeamCity plugin: The [cloudnative webhooks plugin](https://github.com/cloudnative/teamcity-webhooks/) was also installed on the server. Removing it and restarting TeamCity cleared everything up.

I tested 1.1.68.119 and had the same problem. The `<project>/pluginData/plugin-settings.xml` didn't contain any webhooks elements at all, which made me suspicious because I know there were some webhooks set up on that project in the past. At this point I checked for other plugins and discovered that the cloudnative webhooks plugin was installed.
